tests/unit/s2n_connection_blinding_test.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+
+#define ONE_S INT64_C(1000000000)
+
+S2N_RESULT s2n_connection_calculate_blinding(struct s2n_connection *conn, int64_t *min, int64_t *max);
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    struct {
+        uint32_t custom_blinding;
+        uint64_t expected_min;
+        uint64_t expected_max;
+    } test_cases[] = {
+        { .custom_blinding = 0, .expected_min = 0, .expected_max = 0 },
+        { .custom_blinding = 1, .expected_min = ONE_S / 3, .expected_max = ONE_S },
+        { .custom_blinding = 3, .expected_min = (3 * ONE_S) / 3, .expected_max = 3 * ONE_S },
+        { .custom_blinding = 5, .expected_min = (5 * ONE_S) / 3, .expected_max = 5 * ONE_S },
+        { .custom_blinding = 10, .expected_min = (10 * ONE_S) / 3, .expected_max = 10 * ONE_S },
+        { .custom_blinding = 15, .expected_min = (15 * ONE_S) / 3, .expected_max = 15 * ONE_S },
+        { .custom_blinding = 100, .expected_min = (100 * ONE_S) / 3, .expected_max = 100 * ONE_S },
```

Comment:
Most of these test cases aren't really interesting. You're repeating the math in order to write the expected_min and expected_max anyway.

We probably just need 0, 1 (because you'll get a min < 1s), 1<x<30 (happy case), 30 (to verify it matches the default min and max), and UINT32_MAX. I'd also maybe suggest hard coding the expected values rather than calculating them.